Makefile

```diff
@@ -1,55 +1,50 @@
-VIRTUALENV = virtualenv
 VENV := $(shell echo $${VIRTUAL_ENV-$$PWD/.venv})
 PYTHON = $(VENV)/bin/python
-DEV_STAMP = $(VENV)/.dev_env_installed.stamp
 INSTALL_STAMP = $(VENV)/.install.stamp
-TEMPDIR := $(shell mktemp -d)
 
 .IGNORE: clean
-.PHONY: all install virtualenv tests install-dev tests-once
+.PHONY: all install virtualenv tests tests-once
 
 OBJECTS = .venv .coverage
 
 all: install
-install: $(INSTALL_STAMP)
-$(INSTALL_STAMP): $(PYTHON) requirements.txt setup.py
+
+install: $(INSTALL_STAMP) pyproject.toml requirements.txt
+$(INSTALL_STAMP): $(PYTHON) pyproject.toml requirements.txt
 	$(VENV)/bin/pip install -r requirements.txt
-	$(PYTHON) setup.py develop
+	$(VENV)/bin/pip install ".[dev]"
```

Comment:
If I write `pip install ".[dev] -c requirements.txt` I get into a loop of version on boto3/botocore 🤔 

```
...
Collecting boto3 (from kinto-signer)
  Using cached boto3-1.26.159-py3-none-any.whl.metadata (7.0 kB)
Collecting botocore<1.30.0,>=1.29.159 (from boto3->kinto-signer)
  Using cached botocore-1.29.159-py3-none-any.whl.metadata (5.9 kB)
...
```

Reply:
Can't reproduce, did you fix it already?

Reply:
```diff
--- a/Makefile
+++ b/Makefile
@@ -11,8 +11,7 @@ all: install
 install: $(INSTALL_STAMP) pyproject.toml requirements.txt
 $(INSTALL_STAMP): pyproject.toml requirements.txt
        python -m venv $(VENV)
-       $(VENV)/bin/pip install -r requirements.txt
-       $(VENV)/bin/pip install ".[dev]"
+       $(VENV)/bin/pip install ".[dev]" -c requirements.txt
        touch $(INSTALL_STAMP)

```

I can reproduce, it loops through hundreds of versions of botocore and boto

https://gist.github.com/leplatrem/5a3c3d564f2970c1fc135a67b29b6ccd